### PR TITLE
Task 7: Lambda Authorizer

### PR DESCRIPTION
**Pull request by MaxPlav**
### Frontend
Pull Request - https://github.com/MaxPlav/nodejs-aws-fe/pull/5
Cloud Front - https://d2l7k9u8jmnbqh.cloudfront.net/

### Links
- "Import" endpoint covered with authorization lambda: https://5ylpe6sy93.execute-api.eu-west-1.amazonaws.com/dev/import?name=myfile.csv
- Cognito login page for the shop: https://node-army-shop.auth.eu-west-1.amazoncognito.com/login?client_id=21h3i88k07gk7ol29apcduhlrv&response_type=token&scope=aws.cognito.signin.user.admin+email+openid+phone+profile&redirect_uri=https://d2l7k9u8jmnbqh.cloudfront.net/
- "Get all products" endpoint accessible only with "Authorization" token: https://dnlo58ogw8.execute-api.eu-west-1.amazonaws.com/dev/products **Doesn't work without token!**

### What was done
- authorization-service is added to the repo, has correct _basicAuthorizer_ lambda and correct serverless.yaml file **done**
- import-service serverless.yaml file has authorizer configuration for the _importProductsFile_ lambda. Request to the importProductsFile lambda should work only with correct authorization_token being decoded and checked by _basicAuthorizer_ lambda. The response should be in 403 HTTP status if access is denied for this user (invalid authorization_token) and in 401 HTTP status if "Authorization" header is not provided. **done**
- update client application to send Authorization: Basic authorization_token header on import. Client should get authorization_token value from browser localStorage authorization_token = localStorage.getItem('authorization_token') **done**
- Client application should display alerts for the responses in 401 and 403 HTTP statuses. This behavior should be added to the nodejs-aws-fe-main/src/index.tsx file **done**
- Add Login page and protect _getProducts_ lambda by the Cognito Authorizer **done**

### CODE REVIEW

File: authorization-service/dotenv.config.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const dotenv = require('dotenv');
```
Comment (Anton-Koniouchevsky) [inline authorization-service/dotenv.config.js:1]: it's possible to pass env variables to the application even without the 'dotenv' module. Serverless supports env files out of the box - https://www.serverless.com/framework/docs/environment-variables/

File: authorization-service/src/functions/basicAuthorizer/handler.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    const plainCreds = buff.toString('utf-8').split(':');
+
+    const username = plainCreds[0];
+    const password = plainCreds[1];
+
+    console.log(`username=${username} and password=${password}`);
+    const storedUserPassword = process.env[username];
+
+    if (!storedUserPassword) {
+      cb('Access Denied');
```
Comment (Anton-Koniouchevsky) [inline authorization-service/src/functions/basicAuthorizer/handler.ts:34]: For any token value other than "unauthorized", the authorizer returns an HTTP 500 status code. https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-use-lambda-authorizer.html

File: authorization-service/dotenv.config.js (hunk 1 above)
Comment (MaxPlav) [reply authorization-service/dotenv.config.js:1]: ok, thanks

File: authorization-service/src/functions/basicAuthorizer/handler.ts (hunk 2 above)
Comment (MaxPlav) [reply authorization-service/src/functions/basicAuthorizer/handler.ts:34]: ok, thanks

**Pull request merged**